src/chunkserver/trash.cpp

```diff
@@ -303,25 +306,28 @@ bool Trash::IsWALFile(const std::string &fileName) {
 }
 
 uint32_t Trash::CountChunkNumInCopyset(const std::string &copysetPath) {
-    std::string dataPath = copysetPath + "/" + RAFT_DATA_DIR;
-    std::vector<std::string> chunks;
-    localFileSystem_->List(dataPath, &chunks);
-
-    uint32_t chunkNum = 0;
-    // 遍历data下面的chunk
-    for (auto &chunk : chunks) {
-        // 不是chunkfile或者snapshotfile
-        if (!IsChunkOrSnapShotFile(chunk)) {
-            LOG(WARNING) << "Trash find a illegal file:"
-                         << chunk << " in " << dataPath
-                         << ", filename: " << chunk;
-            continue;
+    auto count = [&](const std::string& path) {
+        std::vector<std::string> chunks;
+        localFileSystem_->List(path, &chunks);
+
+        uint32_t chunkNum = 0;
+        for (auto& chunk : chunks) {
+            // valid: chunkfile, snapshotfile, walfile
+            if (!(IsChunkOrSnapShotFile(chunk) || IsWALFile(chunk))) {
```

Comment:
if we both use walpool and chunkfilepool, the 'wal in trash' and the 'chunk in trash'  should be two field.

Reply:
ditto